core/src/main/java/org/apache/accumulo/core/file/blockfile/impl/CachableBlockFile.java

```diff
@@ -342,7 +342,11 @@ private BlockRead getBlock(String _lookup, BlockCache cache, BlockLoader loader)
 
     private BlockRead cacheBlock(String _lookup, BlockCache cache, BlockReader _currBlock, String block) throws IOException {
 
-      if ((cache == null) || (_currBlock.getRawSize() > cache.getMaxSize())) {
+      // ACCUMULO-4716 - Define MAX_ARRAY_SIZE smaller than Integer.MAX_VALUE to prevent possible OutOfMemory
+      // error as described in stackoverflow post: https://stackoverflow.com/a/8381338.
+      int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
```

Comment:
Personally, I would make this a static constants because all caps usually implies that by convention.